### PR TITLE
 Desktop: Show installed plugins in Help - About Joplin #6143 

### PR DESCRIPTION
**Pull request by Taniya003**
solves#6143

All the installed plugins will be visible in the About Joplin dialog box. Below is the screenshot :

      
Screenshot from 2022-02-26 15-05-57

### CODE REVIEW

File: packages/app-desktop/gui/MenuBar.tsx [hunk 1]
```diff
@@ -92,6 +94,26 @@ interface Props {
...
+			manifest: plugin.manifest,
+			devMode: plugin.devMode,
+		});
+	}
+	output.sort((a: PluginItem, b: PluginItem) => {
+		return a.manifest.name < b.manifest.name ? -1 : +1;
+	});
+	return output;
+}
+
```
Comment (laurent22) [inline packages/app-desktop/gui/MenuBar.tsx:116]: remove this copy and pasted code

File: packages/app-desktop/gui/MenuBar.tsx [hunk 2]
```diff
@@ -421,8 +443,18 @@ function useMenu(props: Props) {
...
+				const pluginService = PluginService.instance();
+				const pluginItems = useInstalledPlugins(pluginService.plugins);
+				function random(pluginItems: PluginItem[]): string {
+					let content = v.message.concat('\n');
+					for (const item of pluginItems) {
+						content = content.concat(`\n${item.manifest.name}: ${item.manifest.version}`);
+					}
+					return content;
+				}
+				const message = random(pluginItems);
```
Comment (laurent22) [inline packages/app-desktop/gui/MenuBar.tsx:455]: why is it called random? Just sort and append the plugin name and version and that's it.

**Comment by Daeraxa** [comment]
One of the concerns in the original issue is what happens if you have loads of plugins installed? Do weird things happen to the about box? (I think there are around 100 plugins now).

File: packages/app-desktop/gui/MenuBar.tsx [hunk 3]
```diff
@@ -421,8 +422,31 @@ function useMenu(props: Props) {
...
+						});
+					}
+					if (items) {
+						for (const item of items) {
+							content = content.concat(`\n${item.name}: ${item.version}`);
+						}
+					}
+
+					return content;
+				}
```
Comment (laurent22) [inline packages/app-desktop/gui/MenuBar.tsx:446]: Honestly this code is a joke. I don't even know things like `pluginService[plugins]` can work and all you are doing is mapping a structure to the same structure, so you're doing nothing really. To top it up you're sorting the array within the loop and check if the "items" variable is set when it obviously always is.

This is a project used by many people, not a place for your to learn how to code. So I'm closing this, and please do not open a new pull request.

**Comment by laurent22** [comment]
> One of the concerns in the original issue is what happens if you have loads of plugins installed? Do weird things happen to the about box? (I think there are around 100 plugins now).

That's a good point. I've added that to the original issue. We'll just truncate the list if it's too long, but we'll still include everything when the user presses "Copy".

**Pull request closed without merging**